movement/watch_faces/complication/counter_face.c

```diff
@@ -75,22 +94,46 @@ bool counter_face_loop(movement_event_t event, movement_settings_t *settings, vo
 
 // beep counter index times
 void beep_counter(counter_state_t *state) {
-	
 	int low_count = state->counter_idx/5;
 	int high_count = state->counter_idx - low_count * 5; 
-	
-	for (int i=0; i<low_count; i++) {
-		watch_buzzer_play_note(BUZZER_NOTE_A6, 50);
-	    watch_buzzer_play_note(BUZZER_NOTE_REST, 100);
-	}
-	
-	//sleep between high and low
-    watch_buzzer_play_note(BUZZER_NOTE_REST, 200);
-	
-	for (int i=0; i<high_count; i++) {
-		watch_buzzer_play_note(BUZZER_NOTE_B6, 50);
-	    watch_buzzer_play_note(BUZZER_NOTE_REST, 100);
-	}
+	int8_t sound_seq[] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+    int i=0;
+    if (low_count > 0) { 
+        sound_seq[i] = BUZZER_NOTE_A6;
+        i++;
+        sound_seq[i] = 3;
+        i++;
+        sound_seq[i] = BUZZER_NOTE_REST;
+        i++;
+        sound_seq[i] = 6;
+        i++;
+        if (low_count > 1) {
+            sound_seq[i] = -2;
+            i++;
+            sound_seq[i] = low_count-1;
+            i++;
+        }
+        sound_seq[i] = BUZZER_NOTE_REST;
+        i++;
+        sound_seq[i] = 6;
+        i++;
+    }
+    if (high_count > 0) {
+        sound_seq[i] = BUZZER_NOTE_B6;
+        i++;
+        sound_seq[i] = 3;
+        i++;
+        sound_seq[i] = BUZZER_NOTE_REST;
+        i++;
+        sound_seq[i] = 6;
+        i++;
+    }
+    if (high_count > 1) {
+        sound_seq[i] = -2;
+        i++;
+        sound_seq[i] = high_count-1;
+    }
+    watch_buzzer_play_sequence((int8_t *)sound_seq, NULL);
```

Comment:
sound_seq is a stack variable, and this function goes out of scope as soon as `watch_buzzer_play_sequence()` returns. This means that while the audio is playing (asynchronously), sound_seq can be overwritten by any future function calls pushing things onto the stack. `sound_seq` should probably be static, but also keep in mind then that your initialization will only take effect on the first call of `beep_counter()` so you may want to `memset()` it.

Reply:
Thanks for your feedback! I implemented the suggested changes in a new commit.